src/ui/PickColor.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react'
+import { HexColorPicker } from 'react-colorful'
+import { setState, useStore } from '../store'
+
+export function PickColor(): JSX.Element {
+  const [color, setColor] = useState(useStore((state) => state.color))
```

Comment:
Maybe it'll be clearer if I suggest it for this line instead.

```suggestion
  const setColor = (color: string) => setState({ color })
```

Then you can remove the `useEffect` and get the color from the store.

Reply:
ok, this is what we have now:
```
  const setColor = (color: string) => setState({ color })
  const [color] = useStore((state) => [state.color, state.set])
  const [pickcolor] = useStore((state) => [state.pickcolor, state.set])
  const close = () => setState({ pickcolor: false })
```